statsmodels/stats/multitest.py

```diff
@@ -444,3 +444,241 @@ def fdrcorrection_twostage(pvals, alpha=0.05, method='bky', iter=False,
         return reject, pvalscorr_, ntests - ri, alpha_stages
     else:
         return rej, pvalscorr, ntests - ri, alpha_stages
+
+
+def local_fdr(zscores, null_proportion=1.0, null_pdf=None, deg=7,
+              nbins=30):
+    """
+    Calculate local FDR values for a list of Z-scores.
+
+    Arguments
+    ---------
+    zscores : array-like
+        A vector of Z-scores
+    null_proportion : float
+        The assumed proportion of true null hypotheses
+    null_pdf : function mapping reals to positive reals
+        The density of null Z-scores; if None, use standard normal
+    deg : integer
+        The maximum exponent in the polynomial expansion of the
+        density of non-null Z-scores
+    nbins : integer
+        The number of bins for estimating the marginal density
+        of Z-scores.
+
+    Returns
+    -------
+    fdr : array-like
+        A vector of FDR values
+
+    References
+    ----------
+    B Efron (2008).  Microarrays, Empirical Bayes, and the Two-Groups
+    Model.  Statistical Science 23:1, 1-22.
+
+    Examples
+    --------
+    Basic use (the null Z-scores are taken to be standard normal):
+
+    >>> fdr = local_fdr(zscores)
+
+    Use an empirical null distribution estimated from the data:
+
+    >>> null = EmpiricalNull(zscores)
+    >>> fdr = local_fdr(zscores, null_pdf=null.pdf)
+    """
+
+    from statsmodels.genmod.generalized_linear_model import GLM
+    from statsmodels.genmod.generalized_linear_model import families
+    from statsmodels.regression.linear_model import OLS
+
+    # Bins for Poisson modeling of the marginal Z-score density
+    minz = min(zscores)
+    maxz = max(zscores)
+    bins = np.linspace(minz, maxz, nbins)
+
+    # Bin counts
+    zhist = np.histogram(zscores, bins)[0]
+
+    # Bin centers
+    zbins = (bins[:-1] + bins[1:]) / 2
+
+    # The design matrix at bin centers
+    dmat = np.vander(zbins, deg + 1)
+
+    # Use this to get starting values for Poisson regression
+    md = OLS(np.log(1 + zhist), dmat).fit()
+
+    # Poisson regression
+    md = GLM(zhist, dmat, family=families.Poisson()).fit(start_params=md.params)
+
+    # The design matrix for all Z-scores
+    dmat_full = np.vander(zscores, deg + 1)
+
+    # The height of the estimated marginal density of Z-scores,
+    # evaluated at every observed Z-score.
+    fz = md.predict(dmat_full) / (len(zscores) * (bins[1] - bins[0]))
+
+    # The null density.
+    if null_pdf is None:
+        f0 = np.exp(-0.5 * zscores**2) / np.sqrt(2 * np.pi)
+    else:
+        f0 = null_pdf(zscores)
+
+    # The local FDR values
+    fdr = null_proportion * f0 / fz
+
+    fdr = np.clip(fdr, 0, 1)
+
+    return fdr
+
+
+class EmpiricalNull(object):
```

Comment:
thinking about this some more, `Empirical` would be misleading for me outside of the context. I would think the analog would be ECDF, and an empirical distribution. But here the distribution is fully parametric.
maybe `NullDistribution`